tests/test_decision_making/test_acquisition_functions/test_thompson_sampling.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2023 The GPJax Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from beartype.typing import Callable
+from jax.config import config
+import jax.numpy as jnp
+import jax.random as jr
+from jaxtyping import (
+    Array,
+    Float,
+)
+import pytest
+
+from gpjax.dataset import Dataset
+from gpjax.decision_making.acquisition_functions.thompson_sampling import (
+    ThompsonSampling,
+)
+from gpjax.decision_making.utils import OBJECTIVE
+from gpjax.gps import (
+    ConjugatePosterior,
+    NonConjugatePosterior,
+    Prior,
+)
+from gpjax.kernels import RBF
+from gpjax.likelihoods import (
+    Gaussian,
+    Poisson,
+)
+from gpjax.mean_functions import Zero
+
+config.update("jax_enable_x64", True)
+
+
+def generate_dummy_1d_dataset() -> Dataset:
```

Comment:
For this generator and all below, could you add a seed value argument e.g., `seed_value` that then feeds in the key i.e., `jr.PRNGKey(seed_value)`? All subsequent unit tests could then be validated for more than one seed/dataset.

Reply:
Good idea, was using `linspace` to generate the test points, but will also make this random to improve quality of testing.